MalApi/MyAnimeListApi.cs

```diff
@@ -227,6 +305,147 @@ public Task<MalUserLookupResults> GetAnimeListForUserAsync(string user)
         }
 
         /// <summary>
+        /// Updates a user's anime list entry.
+        /// </summary>
+        /// <param name="animeId">ID of the anime</param>
+        /// <param name="updateInfo">The updated information</param>
+        /// <param name="cancellationToken"></param>
+        /// <param name="user">MAL user</param>
+        /// <param name="password">MAL password</param>
+        /// <returns></returns>
+        public async Task<string> UpdateAnimeForUserAsync(int animeId, UpdateAnimeObject updateInfo, string user, string password, CancellationToken cancellationToken)
+        {
+            string userInfoUri = string.Format(MalAppAnimeUpdateUri, Uri.EscapeDataString(animeId.ToString()));
+
+            Logging.Log.InfoFormat("Updating anime entry for MAL anime ID {0}, user {1} using URI {2}", animeId, user, userInfoUri);
+
+            Func<string, string> responseProcessingFunc = (response) =>
+            {
+                using (TextReader xmlTextReader = new StringReader(response))
+                {
+                    try
+                    {
+                        return xmlTextReader.ToString();
```

Comment:
If responseProcessingFunc has nothing useful to return, which makes sense for an update API call, responseProcessingFunc should have a return type of object and return null.